plugins/modules/cloud/misc/proxmox_disk.py

```diff
@@ -0,0 +1,719 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2022, Castor Sky (@castorsky) <csky57@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: proxmox_disk
+short_description: Management of a disk of a Qemu(KVM) VM in a Proxmox VE cluster.
+version_added: 5.5.0
+description:
+  - Allows you to perform some supported operations on a disk in Qemu(KVM) Virtual Machines in a Proxmox VE cluster.
+author: "Castor Sky (@castorsky) <csky57@gmail.com>"
+options:
+  name:
+    description:
+      - The unique name of the VM.
+      - You can specify either I(name) or I(vmid) or both of them.
+    type: str
+  vmid:
+    description:
+      - The unique ID of the VM.
+      - You can specify either I(vmid) or I(name) or both of them.
+    type: int
+  disk:
+    description:
+      - The disk key (C(unused[n]), C(ide[n]), C(sata[n]), C(scsi[n]) or C(virtio[n])) you want to operate on.
```

Comment:
@castorsky it just occurred to me that it might be interesting to publish those ranges in lines 429-435 so that the users are aware of the limits.

Reply:
I've checked official Proxmox docs - it has slight mentions about ranges (and they are not match with API ranges). So it will be surely useful info, will add.